### ISSUE TITLE
Use the user set voice assistant for voice search in app and widget

### ISSUE THREAD

**Issue by SeburoSUMO** [mozilla-mobile/fenix#12702]
### What is the user problem or growth opportunity you want to see solved?
If Fenix followed the user choice for assist apps, it would be a) supporting the user's choice and b) better able to support other voice search tools (that may include any future work that comes out of the Mozilla Common Voice project).

### How do you know that this problem exists today? Why is this important?
At present, Fenix voice searches are carried out by a Google voice search tool.  Users may wish to use a different voice search tool and have set that in the "Assist app" preferences, but when they do, Fenix still defaults to a Google voice search experience.

### Who will benefit from it?
Users wishing to use voice search in Fenix.

**Comment by madb1lly** [comment]
Hi @SeburoSUMO,
I did some testing on this because I was intrigued to know what works and what doesn't. I tested on a Nokia 8 with Android 9.

In fact, Fenix seems to use whatever service is set as the "Voice Input" in Settings => Apps and Notifications => Default Apps => Assist and Voice Input => Voice Input.  This need not be Google, but it is very difficult to find other voice input services available. I tried Konele and it worked.

However, Fenix only makes the Voice Search microphone button available when Google is chosen as the default search provider, which makes it appear that Google Voice Search is being used. I am pretty sure this is not the case because I can select another search provider after using the voice search function.

Since Google doesn't need to be chosen as the default search provider in order for Voice Search to be used I have raised a bug about this: #15348.

Cheers 🙂
